packages/website/src/hooks/useClipboard.ts

```diff
@@ -0,0 +1,17 @@
+import { useCallback } from 'react';
+
+import { useDebouncedToggle } from './useDebouncedToggle';
+
+export type useClipboardResult = [copied: boolean, copy: () => void];
+
+export function useClipboard(code: () => string): useClipboardResult {
+  const [copied, setCopied] = useDebouncedToggle<boolean>(false);
+
+  const copy = useCallback(() => {
+    void navigator.clipboard.writeText(code()).then(() => {
```

Comment:
[Edge case] Aha, this is why I don't like `void` for `no-floating-promises` - it masks real issues!

Sometimes browsers fail to copy to the clipboard. Most of those times are from users not giving them permission; once in a while it's an OS issue. Either way it'd be good to give them an indication that it failed.

I think it's fine to file this as a followup issue - but it'd be nice to have in this PR.

Reply:
i don't think we have a way to display and error's like that, unless we add some sort of toast, i can catch and `console.error` but there will be no difference